### PR TITLE
Block-based checkout - Shipping section changes

### LINKED ISSUE #43849: "Shipping" and "Local Pickup" text in the Checkout block page editor fail to display correctly
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

"Shipping" and "Local Pickup" text in the Checkout block page editor fail to display correctly due to color contrast when hovered over with recent Gutenberg.

https://github.com/woocommerce/woocommerce/assets/11503784/2dd6fb1f-012e-489b-a9c3-63b5626b59bd

### Expected behavior

"Shipping" and "Local Pickup" text should be displayed correctly when hovered over. 

### Actual behavior

"Shipping" and "Local Pickup" text are not visible correctly when hovered over. 

### Steps to reproduce

1. Activate the latest Gutenberg plugin. 
2. Enable local up. 
3. Navigate to the Checkout page editor. 
4. Hover the shipping and local pickup options

### WordPress Environment

Latest

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### LINKED ISSUE #46013: [Enhancement]: Block-based checkout - Shipping section changes
This is a subticket of https://github.com/woocommerce/woocommerce/issues/44835, which has more context on the changes

- [x] Rename `Shipping method` to `Delivery`, see 2JyXNimI4oGeJZhgQNZE9V-fi-2957_13784

CleanShot 2024-03-04 at 14 55 02@2x

- [x] Restyle and rename tabs from “Shipping” and “Local pickup” to “Ship” & “Pickup” (picture above)

- [ ] ~"Country" field should be first in the address form. This is especially important if geolocation is off. Many fields in checkout country specific conditions (picture above again)~ → moved to #46513

- [ ] ~Hide address line 2 behind a link. [This is a Baymard recommendation](https://baymard.com/blog/address-line-2) to hide optional and infrequently used fields~ → moved to #46512

CleanShot 2024-03-04 at 14 55 40@2x

### PR DESCRIPTION
**Pull request by nielslange**
### Changes proposed in this Pull Request:

Closes #46013.
Closes #43849.

This PR introduces the following changes:

- Change the title from `Shipping method` to `Delivery` when local pickup is enabled.
- Change the wording from `Shipping` to `Ship`  within the `Shipping Method` selector.
- Change the wording from `Local Pickup` to `Pickup` within the `Shipping Method` selector.
- Display SVG and text horizontally instead of vertically within the `Shipping Method` selector
- Hide local pickup costs by default.
- Ensure that the `Shipping` to `Ship` within the `Delivery` section in the editor are correctly styled when Gutenberg is activated.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions.

## Rename `Shipping Method` section title

1. Go to `/wp-admin/admin.php?page=wc-settings&tab=shipping&section=pickup_location`.
2. Enable local pickup.
3. Go to the frontend.
4. Add a product to the cart.
5. Go to the checkout page.
6. Verify that the `Shipping Method` section title shows `Delivery`

<table>
<tr>
<td valign="top">Before:
<br><br>
Screenshot 2024-04-01 at 14 26 46
</td>
<td valign="top">After:
<br><br>
Screenshot 2024-04-01 at 13 24 28
</td>
</tr>
</table>

## Rename shipping method titles

1. Go to `/wp-admin/admin.php?page=wc-settings&tab=shipping&section=pickup_location`.
2. Enable local pickup.
3. Go to the frontend.
4. Add a product to the cart.
5. Go to the checkout page.
6. Verify that the `Shipping method` selector `Shipping` shows `Ship`.
7. Verify that the `Shipping method` selector `Local Pickup` shows `Pickup`.
8. Verify that the `Shipping method` selector SVG and title are displayed horizontally.

<table>
<tr>
<td valign="top">Before:
<br><br>
Screenshot 2024-04-01 at 14 26 46
</td>
<td valign="top">After:
<br><br>
Screenshot 2024-04-01 at 13 24 28
</td>
</tr>
</table>

## Hide local pickup costs by default

1. Go to `/wp-admin/admin.php?page=wc-settings&tab=shipping&section=pickup_location`.
2. Enable local pickup.
3. Create a test page and add the Checkout block to it.

> [!NOTE]
> If the `Delivery` is visible by default, then refresh the page.

4. Select the Delivery block.
5. Open the sidebar.
6. Verify that the option `Show costs` is not activated by default.
7. Go to the frontend.
8. Add a product to the cart.
9. Go to the checkout.
10. Verify that no prices appear within the Delivery block.

## Fix incorrect styles in editor

1. Install and activate Gutenberg.
2. Go to `/wp-admin/admin.php?page=wc-settings&tab=shipping&section=pickup_location`
3. Enable local pickup.
4. Create a test page and add the Checkout block to it.
5. Verify that the styles of `Ship` and `Pickup` within the `Delivery` section look like the styles when Gutenberg is not activated.

<table>
<tr>
<td valign="top">Before:
<br><br>
Screenshot 2024-04-29 at 14 50 57
</td>
<td valign="top">After:
<br><br>
Screenshot 2024-04-29 at 14 50 27
</td>
</tr>
</table>

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Update delivery titles & pickup options; refine shipping selector layout & address field visibility in the Checkout block.

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/form.tsx [hunk 1]
```diff
@@ -72,6 +73,17 @@ const Form = < T extends AddressFormValues | ContactFormValues >( {
 		};
 	}, [ currentFields, currentFieldConfig, currentCountry, addressType ] );
 
+	const [ isAddress2Visible, setIsAddress2Visible ] = useState(
+		objectHasProp( values, 'address_2' ) && values.address_2 !== ''
+	);
+
+	const toggleAddress2Visibility = (
```
Comment (opr) [inline plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/form.tsx]: Consider using [`useCallback`](https://react.dev/reference/react/useCallback) here to cache this function between renders

Comment (nielslange) [reply plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/form.tsx]: Thanks for suggesting this, @opr. I've just updated the PR accordingly.

**Comment by dominique-vias** [comment]
For the `Hide Address 2 behind toggle` change, here is the corresponding design proposal for the editor side:

- [ ] Changed address toggle label to `Address line 2`
- [ ] Replace required checkbox to `Optional` and `Required` radio selections. This allows us to communicate our recommendations.

<table>
<tr>
<td valign="top">Before:
<br><br>
Before
</td>
<td valign="top">After:
<br><br>
After
</td>
</tr>
</table>

### Front-end implementation of the settings

Out of the box experience should be:

- Company off
- Address line 2 on + optional
- Phone on + optional

Design

**Review by senadir — APPROVED** [review]
Looking good and testing good from my end

**Comment by nielslange** [comment]
Thanks for reviewing this PR, @senadir. 🙌

**Pull request merged**